### DISCUSSION TITLE
SG Ready Support

### DISCUSSION THREAD

**Discussion by wersim** [marq24/ha-senec-v3#102] [Ideas]
Hi @marq24,

is it possible to support the SG Ready Interface via Web API?
![image](https://github.com/user-attachments/assets/9b1814c2-0915-4730-81e2-3caab62e79af)

If there is anything I could provide please let me know.

**Comment by marq24** [comment]
Ja generell könnte man das unterstützen - zumal ich siner Zeit auch geglaubt habe, so was wie SG-Ready müsste man haben... Allerdings, WIE oft ändert man schon diese Werte?!

Ich verwende einfach HA Automatisierungen für alles was ich mit PV Überschuss steuern möchte...

**Reply by wersim** [reply]
Das stimmt, die Grenzwerte ändert man wohl eher nicht so oft. Mir würde es eher um eine Überwachung des Status gehen, denn ich hatte es bereits zweimal erlebt, dass der Senec-Speicher auch dann noch der Wärmepumpe eine Anlaufempfehlung sendet, wenn die derzeit eingespeiste Überschussenergie deutlich unter den Schwellwerten liegt.
Ich dachte daher daran, in HA die aktuelle Einspeisung sowie die Statusänderungen zu tracken, und SG Ready evtl. automatisiert zu deaktivieren, falls die Überschusseinspeisung nicht ausreicht.

![image](https://github.com/user-attachments/assets/ed0096b1-3a83-48e6-b7e4-fda17ea55325)

**Comment by marq24** [comment]
dazu müsstest/könntest Du mir mal die unterschiedlichen "SwitchState.MODE_" mitteilen - und was die MODE1...MODEX tatsächlich bedeuten... Also das was im Web unter "Zustand" angezeight wird, in der App dann aber als "Status" steht...

also "MODE3" scheint "Anlaufempfehlung" zu sein... aber die übigen modes benötige ich dann auch noch...

**Reply by wersim** [reply]
Es existieren vier Zustände:
MODE1 = Sperre
MODE2 = Normalbetrieb
MODE3 = Anlaufempfehlung
MODE4 = Anlaufbefehl    

Hier noch der entsprechende Auszug aus einer Senec Anleitung zur Wärmepumpensteuerung:
![image](https://github.com/user-attachments/assets/d3569df6-538e-4211-9a73-d95b6c5b4d84)

**Comment by marq24** [comment]
Please check the new Version 2024.8.5

**Reply by wersim** [reply]
Thanks @marq24, 
successfully tested the switch and I'll added new sg ready entities to the HA recorder. Happy that I can track the states now! I'm sure that will give me some more insights how the SG Ready interface operates...

**Reply by wersim** [reply]
Thanks to the SG Ready support in the new version, you can now see that SENEC has implemented the SG Ready interface incorrectly. Even if the feed-in value has fallen below the threshold of 3100 watts to 0 watts for over an hour, SENEC still sends a start-up command to the heat pump.... until feature is switched off manually. :(
![Screenshot_20240831-203607](https://github.com/user-attachments/assets/0a02e804-386e-4242-8156-217ab0d3abd9)

**Reply by marq24** [reply]
What should I say? [well first of all thank you for paypalme] - but what else? Also es ist jetzt nicht wirklich eine Überraschung, das SENEC die einfachsten Dinge nicht wirklich auf die Kette bringt - bzw. die Dinge einfach mal selbst TESTED... 

Einfach traurig - wie gesagt, ich habe den Luxus, dass ich meine Wärmepumpe komplett über HTTP ansteuern kann und gar nicht auf SG-Ready angewiesen bin (ich weis das hilft Dir natürlich nicht weiter)... Bin aber dann doch auch Neugierig und möchte gerne erfahren, wann SENEC dann meint, das es jetzt gut wäre mit dem SGReady Signal...
